src/ansys/fluent/core/session.py

```diff
@@ -308,16 +308,31 @@ def force_exit(self) -> None:
         data."""
         self._fluent_connection.force_exit()
 
-    def upload(self, file_name: str):
+    def _warning(self, method_name: str) -> str:
+        """User warning for upload/download methods."""
+        return f"You have directly called the {method_name} method of the session. \
+        Please be advised that for the current version of Fluent, many API methods \
+        automatically handle file uploads and downloads internally. You may not \
+        need to explicitly call `upload()` or `download()` in most cases. \
+        However, there are exceptions, particularly in PMFileManagement, where complex \
+        file interactions require explicit use of `upload()` and `download()` methods \
+        for relevant files."
+
+    def upload(
+        self, file_name: Union[list[str], str], remote_file_name: Optional[str] = None
+    ):
         """Upload a file to the server.
 
         Parameters
         ----------
         file_name : str
             Name of the local file to upload to the server.
+        remote_file_name : str, optional
+            remote file name, by default None
         """
+        warnings.warn(self._warning("upload()"), UserWarning)
```

Comment:
Given that the intention was to factor out all duplication, more could be delegated to the called function.

Reply:
Done. Thank you.